persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/DelegatingAuthorizer.java

```diff
@@ -0,0 +1,222 @@
+import org.apache.cassandra.stargate.exceptions.RequestValidationException;
+import org.javatuples.Pair;
+
+public class DelegatingAuthorizer extends CassandraAuthorizer {
+
+  private AuthorizationProcessor authProcessor;
+
+  public void setProcessor(AuthorizationProcessor processor) {
+    authProcessor = processor;
+  }
+
+  @Override
+  public Set<Permission> grant(
+      AuthenticatedUser performer,
+      Set<Permission> permissions,
+      IResource resource,
+      RoleResource grantee,
+      GrantMode... grantModes) {
+    AuthorizationProcessor processor = authProcessor;
+    if (processor == null) {
+      return super.grant(performer, permissions, resource, grantee, grantModes);
+    }
+
+    Collection<CompletableFuture<Void>> stages = new ArrayList<>(grantModes.length);
+
+    for (GrantMode grantMode : grantModes) {
+      Pair<PermissionKind, AuthorizationOutcome> mode = grantMode(grantMode);
+      CompletionStage<Void> stage =
+          processor.addPermissions(
+              ImmutableActor.of(performer.getName()),
+              mode.getValue1(),
+              mode.getValue0(),
+              permissions(permissions),
+              resource(resource),
+              role(grantee));
+
+      stages.add(stage.toCompletableFuture());
+    }
+
+    get(CompletableFuture.allOf(stages.toArray(new CompletableFuture[0])));
+
+    // assume all permissions were granted if we did not an exception
+    return permissions;
+  }
+
+  @Override
+  public Set<Permission> revoke(
+      AuthenticatedUser performer,
+      Set<Permission> permissions,
+      IResource resource,
+      RoleResource revokee,
+      GrantMode... grantModes) {
+    AuthorizationProcessor processor = authProcessor;
+    if (processor == null) {
+      return super.revoke(performer, permissions, resource, revokee, grantModes);
+    }
+
+    Collection<CompletableFuture<Void>> stages = new ArrayList<>(grantModes.length);
+
+    for (GrantMode grantMode : grantModes) {
+      Pair<PermissionKind, AuthorizationOutcome> mode = grantMode(grantMode);
+      CompletionStage<Void> stage =
+          processor.addPermissions(
+              ImmutableActor.of(performer.getName()),
+              mode.getValue1(),
+              mode.getValue0(),
+              permissions(permissions),
+              resource(resource),
+              role(revokee));
+
+      stages.add(stage.toCompletableFuture());
+    }
+
+    get(CompletableFuture.allOf(stages.toArray(new CompletableFuture[0])));
+
+    // assume all permissions were revoked if we did not an exception
+    return permissions;
+  }
+
+  private static void get(CompletionStage<Void> stage) {
+    try {
+      // wait for completion since the calling API is synchronous
+      stage.toCompletableFuture().get();
+    } catch (ExecutionException e) {
+      Throwable cause = e.getCause();
+      if (cause instanceof RequestValidationException) {
+        throw (RequestValidationException) cause;
```

Comment:
what value do we get by throwing the `RequestValidationException` in that case instead of coalescing all catch blocks and throw:
`throw new AuthRequestExecutionException(SERVER_ERROR, e.getMessage(), e);` always?